File: src/main/java/com/github/stars/controller/SimilarRepoController.java
package com.github.stars.controller;

import com.github.stars.service.SimilarRepoService;
import org.springframework.web.bind.annotation.*;

import javax.annotation.Resource;
import java.util.*;

@RestController
@RequestMapping("/api/similar")
public class SimilarRepoController {

    @Resource
    private SimilarRepoService similarRepoService;

    /**
     * 发现与指定仓库相似的项目
     */
    @GetMapping("/{repoId}")
    public Map<String, Object> findSimilar(@PathVariable Long repoId) {
        Map<String, Object> result = new LinkedHashMap<>();
        List<SimilarRepoService.SimilarRepo> repos = similarRepoService.findSimilar(repoId);
        result.put("success", true);
        result.put("repos", repos);
        result.put("count", repos.size());
        return result;
    }
}

File: backend-nestjs/src/ai/controllers/similar-repo.controller.ts
import { Controller, Get, Param } from '@nestjs/common';
import { SimilarRepoService } from '../services/similar-repo.service';

@Controller('api/similar')
export class SimilarRepoController {
    constructor(private readonly service: SimilarRepoService) {}

    @Get(':repoId')
    async findSimilar(@Param('repoId') repoId: string) {
        const repos = await this.service.findSimilar(parseInt(repoId));
        return { success: true, repos, count: repos.length };
    }
}
